qfieldsync/core/cloud_converter.py

```diff
@@ -0,0 +1,133 @@
+
+from qfieldsync.libqfieldsync.layer import LayerSource
+from qfieldsync.libqfieldsync.utils.file_utils import copy_images
+
+
+class CloudConverter(QObject):
+    progressStopped = pyqtSignal()
+    warning = pyqtSignal(str, str)
+    task_progress_updated = pyqtSignal(int, int)
+    total_progress_updated = pyqtSignal(int, int, str)
+
+    def __init__(
+        self,
+        project: QgsProject,
+        export_folder: str,
+    ):
+
+        super(CloudConverter, self).__init__(parent=None)
+        self.__max_task_progress = 0
+        self.__convertor_progress = None  # for processing feedback
+        self.__layers = list()
+
+        # elipsis workaround
+        self.trUtf8 = self.tr
+
+        self.export_folder = export_folder
+
+    def convert(self) -> None:  # noqa: C901
+        """
+        Convert the project to a cloud project.
+        """
+
+        project = QgsProject.instance()
+        original_project_path = project.fileName()
+        project_filename = project.baseName()
+
+        try:
+            if not os.path.exists(self.export_folder):
+                os.makedirs(self.export_folder)
+
+            if glob.glob(os.path.join(self.export_folder, "*.qgs")) or glob.glob(
+                os.path.join(self.export_folder, "*.qgz")
+            ):
+                raise Exception(
+                    self.tr("The destination folder already contains a project file")
+                )
+
+            self.total_progress_updated.emit(0, 100, self.trUtf8("Converting project…"))
+            self.__layers = list(project.mapLayers().values())
+
+            # Loop through all layers and copy them to the destination folder
+            pathResolver = QgsProject.instance().pathResolver()
+            for current_layer_index, layer in enumerate(self.__layers):
+                self.total_progress_updated.emit(
+                    current_layer_index,
+                    len(self.__layers),
+                    self.trUtf8("Copying layers…"),
+                )
+
+                layer_source = LayerSource(layer)
+                if not layer_source.is_supported:
+                    project.removeMapLayer(layer)
+                    continue
+
+                if layer.dataProvider() is not None:
+                    md = QgsProviderRegistry.instance().providerMetadata(
+                        layer.dataProvider().name()
+                    )
+                    if md is not None:
+                        decoded = md.decodeUri(layer.source())
+                        if "path" in decoded:
+                            path = pathResolver.writePath(decoded["path"])
+                            if path.startswith("localized:"):
+                                # layer stored in localized data path, skip
+                                continue
+
+                if layer.type() == QgsMapLayer.VectorLayer:
+                    if not layer_source.convert_to_gpkg(self.export_folder):
```

Comment:
So there is NO way to keep you pg layer a PG layer, even if it is a public postgis?

Btw, if it fails, would be nice to have some kind of UI indication